core/src/test/java/org/opensearch/sql/expression/datetime/DateSubAndSubDateTest.java

```diff
@@ -0,0 +1,140 @@
+  }
+
+  @Test
+  public void time_limited_by_24_hours() {
+    var res = subdate(LocalTime.MIN, Duration.ofNanos(1));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalTime.MAX, res.datetimeValue().toLocalTime());
+
+    res = date_sub(LocalTime.of(10, 20, 30),
+        Duration.ofHours(20).plusMinutes(50).plusSeconds(7));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalTime.of(13, 30, 23), res.datetimeValue().toLocalTime());
+  }
+
+  @Test
+  public void returns_datetime_when_args_are_date_and_date_interval() {
+    var res = subdate(LocalDate.of(2020, 2, 20), Period.of(3, 11, 21));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalDate.of(2016, 2, 28).atStartOfDay(), res.datetimeValue());
+
+    res = date_sub(LocalDate.of(1961, 4, 12), Period.of(50, 50, 50));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalDate.of(1906, 12, 24).atStartOfDay(), res.datetimeValue());
+  }
+
+  @Test
+  public void returns_datetime_when_args_are_date_and_time_interval() {
+    var res = subdate(LocalDate.of(2020, 2, 20), Duration.ofHours(1).plusMinutes(2));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalDateTime.of(2020, 2, 19, 22, 58), res.datetimeValue());
+
+    res = date_sub(LocalDate.of(1961, 4, 12), Duration.ofHours(9).plusMinutes(7));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalDateTime.of(1961, 4, 11, 14, 53), res.datetimeValue());
+  }
+
+  @Test
+  public void returns_datetime_when_args_are_time_and_date_interval() {
+    // Date based on today
+    var res = subdate(LocalTime.of(1, 2, 0), Period.ofDays(1));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalDate.now().minusDays(1).atTime(LocalTime.of(1, 2, 0)), res.datetimeValue());
+
+    res = date_sub(LocalTime.MIDNIGHT, Period.ofDays(0));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalDate.now().atStartOfDay(), res.datetimeValue());
+  }
+
+  @Test
+  public void returns_datetime_when_first_arg_is_datetime_or_timestamp() {
+    var res = subdate(LocalDateTime.of(1961, 4, 12, 9, 7), Duration.ofMinutes(108));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalDateTime.of(1961, 4, 12, 7, 19), res.datetimeValue());
+
+    res = date_sub(LocalDateTime.of(1961, 4, 12, 9, 7).toInstant(ZoneOffset.UTC),
+        Duration.ofMinutes(108));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalDateTime.of(1961, 4, 12, 7, 19), res.datetimeValue());
+  }
+
+  @Test
+  public void accepts_negative_interval() {
+    var res = subdate(LocalDateTime.of(2020, 10, 20, 14, 42), Duration.ofDays(-10));
+    assertEquals(DATETIME, res.type());
+    assertEquals(LocalDateTime.of(2020, 10, 30, 14, 42), res.datetimeValue());
+    assertEquals(adddate(LocalDateTime.of(2020, 10, 20, 14, 42), Duration.ofDays(10)), res);
+  }
+
+  @Test
+  public void subdate_has_second_signature_but_not_date_sub() {
```

Comment:
might be worthwhile to include the exception in the title of the test?  
`subdate_has_second_signature_but_not_date_sub_throwsExpressionEvaluationException`?